pkg/controller/core/workload_controller.go

```diff
@@ -543,7 +594,11 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 	}
 	// at this point we know a Workload has at least one Retry AdmissionCheck
 	message := "At least one admission check is false"
-	if err := workload.Evict(ctx, r.client, r.recorder, wl, kueue.WorkloadEvictedByAdmissionCheck, message, "", r.clock); err != nil {
+	if err := workload.Evict(ctx, r.client, r.recorder, wl, kueue.WorkloadEvictedByAdmissionCheck, message, "", r.clock, workload.WithCustomPrepare(func() (*kueue.Workload, error) {
+		// Set Requeued=false so that when the backoff elapses, we can set it to true
+		workload.SetRequeuedCondition(wl, kueue.WorkloadEvictedByAdmissionCheck, "Evicted due to admission check retry", false)
```

Comment:
this makes sense. Did we set it before from another place?

If no it sounds like it was a bug, surprising it didn't backfire more.

if you also think this is a bug we could submit in separete PR we could backport.

Reply:
sure, here's the one: https://github.com/kubernetes-sigs/kueue/pull/7673